src/oc_erchef/apps/chef_objects/src/chef_node.erl

```diff
@@ -159,7 +159,12 @@ ejson_for_indexing(#chef_node{name = Name, environment = Environment}, Node) ->
                                    {<<"chef_environment">>, Environment},
                                    {<<"recipe">>, extract_recipes(RunList)},
                                    {<<"role">>, extract_roles(RunList)},
-                                   {<<"run_list">>, RunList}]),
+                                   {<<"run_list">>, RunList},
+                                   {<<"cheftoplevel_name">>, Name},
```

Comment:
Avoiding the longer name would be nice but I tested your suggestion and I got the following error.

```
knife search "#chef_environment:prod" -i
ERROR: knife search failed: invalid search query: '#chef_environment:prod'
```

The other reason for the longer name was to make it less likely that a node would have an identically named attribute.